extractor/filesystem/os/peversion/peversion.go

```diff
@@ -0,0 +1,330 @@
+	metadata = make(map[string]string)
+
+	peFile, err := pe.New(exePath, &pe.Options{})
+	if err != nil {
+		// PE parsing failed, try filename heuristics.
+		version, prodName = extractVersionFromFilename(exePath)
+		return
+	}
+	defer peFile.Close()
+
+	// Parse the PE file structure.
+	if err = peFile.Parse(); err != nil {
+		version, prodName = extractVersionFromFilename(exePath)
+		return
+	}
+
+	// Extract version resources.
+	versionInfo, err := peFile.ParseVersionResources()
+	if err != nil {
+		version, prodName = extractVersionFromFilename(exePath)
+		return
+	}
+
+	// Extract version: prefer ProductVersion, fallback to FileVersion.
+	if productVersion, ok := versionInfo["ProductVersion"]; ok && productVersion != "" {
+		version = productVersion
+	} else if fileVersion, ok := versionInfo["FileVersion"]; ok && fileVersion != "" {
+		version = fileVersion
+	}
+
+	// Extract product name: prefer ProductName, fallback to InternalName or OriginalFilename.
+	if productName, ok := versionInfo["ProductName"]; ok && productName != "" {
+		prodName = productName
+	} else if internalName, ok := versionInfo["InternalName"]; ok && internalName != "" {
+		prodName = internalName
+	} else if originalFilename, ok := versionInfo["OriginalFilename"]; ok && originalFilename != "" {
+		prodName = originalFilename
+	}
+
+	// Store additional PE version resource fields for security analysis.
+	interestingFields := []string{
+		"CompanyName", "LegalCopyright", "FileDescription",
+		"OriginalFilename", "InternalName", "LegalTrademarks",
+		"PrivateBuild", "SpecialBuild", "Comments",
+	}
+
+	for _, field := range interestingFields {
+		if value, ok := versionInfo[field]; ok && value != "" {
+			metadata[field] = value
+		}
+	}
+
+	// If PE parsing didn't yield version, try filename heuristics.
+	if version == "" {
+		fallbackVersion, _ := extractVersionFromFilename(exePath)
+		version = fallbackVersion
+	}
+
+	// If no product name found in PE resources, use the base filename as fallback.
+	if prodName == "" {
+		base := filepath.Base(exePath)
+		// Remove file extension to get a cleaner product name.
+		prodName = strings.TrimSuffix(base, filepath.Ext(base))
+	}
+
+	return version, prodName, metadata
+}
+
+// extractVersionFromFilename attempts to extract version from the filename.
+// This is a fallback method when PE resource parsing fails.
+// Note: This method only extracts version, not product name, as filenames
+// are unreliable for product identification.
+func extractVersionFromFilename(exePath string) (version, prodName string) {
+	base := filepath.Base(exePath)
+	lower := strings.ToLower(base)
+
+	// Remove architecture markers and file extensions for version extraction.
+	clean := archRegex.ReplaceAllString(lower, "")
+	clean = strings.TrimSuffix(clean, ".exe")
+	clean = strings.TrimSuffix(clean, ".dll")
+
+	// Try to find a version number pattern.
+	verMatch := verRegex.FindStringSubmatch(clean)
```

Comment:
I'm not sure I agree with these rules, do you have real world examples of these versions in names?